### PR TITLE
clang_tot_upgrade

### PR DESCRIPTION
**Pull request by AlexVlx**
Adopt short-form GFX_ip family names in the CMake and Driver infrastructure.

### CODE REVIEW

**Review by scchan — CHANGES_REQUESTED** [review]
Could you rebase your commit since it also includes other commits already in clang_tot_upgrade?

File: CMakeLists.txt [hunk 1]
```diff
@@ -168,32 +170,22 @@ endif (HAS_ROCM EQUAL 0)
...
-  if (AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:8:0:1")
+  elseif (AMDGPU_GPU_TARGET STREQUAL "gfx801")
     list(APPEND AMDGPU_TARGET_LIST "carrizo")
-  endif()
-
-  if (AMDGPU_GPU_TARGET STREQUAL "AMD:AMDGPU:8:0:3")
+  elseif (AMDGPU_GPU_TARGET STREQUAL "gfx802")
+    list(APPEND AMDGPU_TARGET_LIST "tonga")
+  elseif (AMDGPU_GPU_TARGET STREQUAL "gfx803")
     list(APPEND AMDGPU_TARGET_LIST "fiji")
```
Comment (scchan) [inline CMakeLists.txt:202]: Let's keep the support for the long names for now

Comment (scchan) [reply CMakeLists.txt:202]: Maybe add a warning that the long names are being deprecated?

Comment (AlexVlx) [reply CMakeLists.txt:202]: Excellent idea, I wish I had thought about that before! Please see the updated form and if it fits what you had in mind. It is not necessarily the most efficient way of doing things, but I presume we will remove this sooner as opposed to later. Thank you for the suggestion.

**Review by whchung — COMMENTED** [review]
We may also need to add "gfx700" into supported list. I removed Kaveri support a while ago, but it seems we still have ROCm users using Kaveri APU for development.

**Comment by AlexVlx** [comment]
Hello @whchung. Thank you for the reply - isn't Kaveri formally deprecated in the context of ROCm? Which would make enabling it here a bit iffy, unless there's been a recent change in that regard. Having said this, I believe that this argues in favour of a redesign of how we handle this particular aspect (selecting the GFXip family), so that we have a single shared list of enabled IP configurations which drives the decisions everywhere, be it CMake of the Driver. I will not have time to implement that this year, but I think it'd be useful to have overall and we might as well figure out what to do about Kaveri by then. Does that sound reasonable? Thank you. Cheers!

**Comment by whchung** [comment]
@AlexVlx to my understanding there are still users who use Clang 3.5-based HCC and using `--amdgpu-target=AMD:AMDGPU:7:0:0`, so in order not to bring them unexpected surprises we should still keep this option in Clang 4.0-based HCC. I've merged this PR and will submit commits to address this later on.

File: lib/clamp-device.in [hunk 2]
```diff
@@ -103,12 +103,14 @@ if [ $KMDUMPLLVM == "1" ]; then
 fi
 
 # select appropriate ROCm-Device-Libs per AMDGPU_TARGET
-if [ $AMDGPU_TARGET == "fiji" ]; then
-  OCLC_ISA_VERSION_LIB="$ROCM_LIB/oclc_isa_version_803.amdgcn.bc"
-elif [ $AMDGPU_TARGET == "hawaii" ]; then
+if [ $AMDGPU_TARGET == "gfx701" ]; then
```
Comment (whchung) [inline lib/clamp-device.in:106]: after testing with end-to-end programs, changes in this file are actually incorrect. I've fixed them in further commits.

**Comment by AlexVlx** [comment]
@whchung thank you for the clarification, that's pretty interesting. Having said that, I recall that we are on record saying that Kaveri support is deprecated. Perhaps we should at least issue a warning to that extent? Otherwise users might have even less pleasant surprises if they actually put a codebase together and an upstream driver change breaks them utterly, which could happen at any point...

**Pull request merged**